homeassistant/components/local_calendar/config_flow.py

```diff
@@ -2,18 +2,44 @@
 
 from __future__ import annotations
 
+import logging
 from typing import Any
 
+import httpx
 import voluptuous as vol
 
 from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.helpers.httpx_client import get_async_client
+from homeassistant.helpers.selector import (
+    DurationSelector,
+    DurationSelectorConfig,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
 from homeassistant.util import slugify
 
-from .const import CONF_CALENDAR_NAME, CONF_STORAGE_KEY, DOMAIN
+from .const import (
+    CONF_CALENDAR_NAME,
+    CONF_CALENDAR_URL,
+    CONF_STORAGE_KEY,
+    CONF_SYNC_INTERVAL,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
 
 STEP_USER_DATA_SCHEMA = vol.Schema(
     {
         vol.Required(CONF_CALENDAR_NAME): str,
+        vol.Optional(CONF_CALENDAR_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Optional(CONF_SYNC_INTERVAL): DurationSelector(
```

Comment:
This needs to be removed from the config flow.  The design standards for how to do updates are here https://developers.home-assistant.io/docs/integration_fetching_data/  -- This is something generic that applies to any entity and not something we want to add to the integration specific configuration flow.

The way we'll want to handle this is:
- Pick a reasonable default for how often to refresh this (not too often)
- Users can add their own automation and a schedule using an action call to `homeassistant.update_entity` to refresh

Reply:
I will follow those instructions. Thank you. I have a lot to learn about coding for home assistant integrations.

Since each remote calendar entity has its own URL, I first thought I should go for [Separate polling for each individual entity](https://developers.home-assistant.io/docs/integration_fetching_data/#separate-polling-for-each-individual-entity), and use the `async_update()` method for polling the data. 

However I believe the `async_update()` method is currently called when the calendar entity needs to update the next event attribute, and I do not think we need to fetch the data *that* frequently. Considering this, having one coordinator class for each entity seems to make more sense. Would you agree? Suggestions are welcome.

Reply:
Great thanks. Very happy to have your contribution, and while there is a learning curve it is quite rewarding to contribute to home assistant so really happy to help here. You have a great start already and reviewers are happy to point out best practices that may not be easy to find.

Yeah doing the polling in each entity is fine.

OK so the calendar entity is a bit special, and so i can explain here how it works. Consider here a scenario for Google calendar: Say every 15 minutes you refresh the cloud calendar, but then a new event shows up.  It starts in 5 minutes, so you also need to handle the state transition from "off" to "on". How this works is the calendar base class will handle this for you. You'd just refresh the calendar event every 15 minutes and the base class will set an additional timer based on `event` for 5 minutes and it will trigger another state change.

See https://github.com/home-assistant/core/blob/545dae2e7f2e1d077ca0724f471e7b0ed9f45aff/homeassistant/components/calendar/__init__.py#L535 for details on how this works.

So yeah doing it per entity as you said is fine.